src/api.rs

```diff
@@ -47,9 +48,47 @@ pub enum Request {
         transform: TransformationSpec,
     },
     UpdateScene(Arc<SceneSpec>),
+    Query(QueryRequest),
     Start,
 }
 
+#[derive(Serialize, Deserialize)]
+#[serde(tag = "query", rename_all = "snake_case")]
+pub enum QueryRequest {
+    WaitForNextFrame { input_id: InputId },
+    Scene,
+    Inputs,
+    Outputs,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum Response {
+    Ok {},
```

Comment:
Is `{}` needed here?

Reply:
yes I want to return `{}` instead of null

Reply:
Oh okay, makes sense